adbms/src/main/scala/de/up/hpi/informationsystems/adbms/record/internal/TypedMapLike.scala

```diff
@@ -0,0 +1,132 @@
+  * a `MapLike`. This is an alternative to overloading `apply` and `get`.
+  *
+  * @tparam K    the type of the keys
+  * @tparam V    the type of the associated values
+  * @tparam This the type of the map itself
+  */
+trait TypedMapLike[K[+_ <: V], V, This <: TypedMapBase[K, V]] { self: This =>
+
+  /** Creates a new instance of `This` using the provided data.
+    * @param cells content of the new `This` instance
+    * @return a new `This` using the new content
+    */
+  protected def ctor(cells: Map[K[V], V]): This
+
+  /** Creates a new iterator over all key/value pairs of this map
+    *
+    *  @return the new iterator
+    */
+  def iterator: Iterator[(K[V], V)]
+
+  /** Adds a key/value pair to this map, returning a new map.
+    *  @param    kv the key/value pair
+    *  @tparam   T the type of the value in the key/value pair.
+    *  @return   a new `This` with the new binding added to it
+    */
+  def +[T <: V](kv: (K[T], T)): This =
+    ctor(data + kv)
+
+  /** Removes a key from this map, returning a new map.
+    *  @param    k the key to be removed
+    *  @return   a new `This` without a binding for `key`
+    */
+  def -[T <: V](k: K[T]): This =
+    ctor(data - k)
+
+  /** Adds a number of elements provided by a traversable object
+    *  and returns a new collection with the added elements.
+    *
+    *  @param xs      the traversable object consisting of key-value pairs.
+    *  @return        a new `This` with the bindings of this map and those from `xs`.
+    */
+  def ++[T <: V](xs: TypedMapBase[K, T]): This =
+    ctor(data.++(xs.iterator))
+
+  /** Creates a new map from this one by removing all keys of another
+    *  collection.
+    *
+    *  @param xs     the collection containing the removed elements.
+    *  @return a new `This` that contains all key/value bindings of the current map
+    *  except the key/value bindings of `elems`.
+    */
+  def --(xs: TypedMapBase[K, V]): This =
+    ctor(data.--(xs.iterator.map(_._1)))
+
+  /** Retrieves the value which is associated with the given key. This
+    *  method invokes the `default` method of the map if there is no mapping
+    *  from the given key to a value. Unless overridden, the `default` method throws a
+    *  `NoSuchElementException`.
+    *
+    *  @param  k   the key
+    *  @tparam T   the type of the value in the key/value binding
+    *  @return     the value associated with the given key, or the result of the
+    *              map's `default` method, if none exists.
+    */
+  def apply[T <: V : ClassTag](k: K[T]): T =
+    data(k) match {
+      case t: T => t
+      case _ => default(k)
+    }
+
+  /** Optionally returns the value associated with a key.
+    *
+    *  @param  k    the key value
+    *  @tparam T    the type of the value in the key/value binding
+    *  @return an option value containing the value associated with `key` in this map,
```

Comment:
The doc style isn't consistent here. Doesn't bother me too much but you consciously updated your style to the gutter aligned scaladoc style so I thought maybe you want to make sure it's all coherent?

Reply:
I just copied most of it from the scala Map-API :laughing: But I will adjust the alignment ;)
(applies also to the other style comments)